### PR TITLE
Fix refreshing state after signing in or out

### PR DESCRIPTION
**Pull request by PatrykSzwer**
For #321

### CODE REVIEW

File: src/BlendingEditor/Startup.cs [hunk 1]
```diff
@@ -19,9 +19,9 @@ public class Startup : IStartup
         public void ConfigureServices(IServiceCollection services)
         {
             services
-                .AddStarcounterAuthorization<DataModels.AuthorizationSettings, UserSession, TicketToSession, User>(ConfigureAuthorization)
                 .AddRouter()
-                .SetMasterPage((provider, info) => MasterPage.GetForViewModel(info.SelectedPageType))
+                .SetMasterPage((provider, info) => MasterPage.GetForViewModel(info.SelectedPageType, provider))
```
Comment (joozek78) [inline src/BlendingEditor/Startup.cs]: please add a comment explaining the importance of the order here

File: src/BlendingEditor/Website/Api/CommitHooks.cs [hunk 2]
```diff
@@ -52,9 +52,16 @@ protected void RefreshSignInState()
                     return;
                 }
 
-                SessionHelper.GetMasterPage(_currentUserProvider.GetCurrentUser() == null
-                    ? Self.GET("/blendingeditor/deny")
-                    : Self.GET("/blendingeditor/partials/layoutgroups"));
+                if (_currentUserProvider.GetCurrentUser() == null)
+                {
+                    var master = SessionHelper.GetMasterPage(Self.GET("/blendingeditor/deny"));
+                    master.Html = "/BlendingEditor/views/MasterPageWithoutLayout.html";
```
Comment (joozek78) [inline src/BlendingEditor/Website/Api/CommitHooks.cs]: please assign a symbol to those strings. Maybe a static field in the `MasterPage` class?

File: src/BlendingEditor/Website/Helpers/SessionHelper.cs [hunk 3]
```diff
@@ -9,8 +10,7 @@ public static MasterPage GetMasterPage(Json partial = null)
         {
             if (!(Session.Ensure().Store[nameof(MasterPage)] is MasterPage master))
             {
-                master = new MasterPage();
-                Session.Current.Store[nameof(MasterPage)] = master;
+                throw new InvalidOperationException("Master page was not found in Store.");
```
Comment (joozek78) [inline src/BlendingEditor/Website/Helpers/SessionHelper.cs]: I'd say "session store" instead of "store"

**Comment by PatrykSzwer** [comment]
I applied all your requested changes, please check.

**Comment by warpech** [comment]
I am getting a weird error. @joozek78 please try to isolate it and report it in a relevant place:

1. Run Blending.Mapper, Blending, SignIn.Mapper, SignIn from the Omni set of apps
2. Log in using the top right corner
3. Refresh the page
4. Log out using the top right corner
5. Without refreshing, log in using the top right corner

Expected:

It should log me in

Actual:

I am getting an error `Could not sign in due to unexpected error. Please contact the administrator`.

It works if I reload the page.

image

**Comment by PatrykSzwer** [comment]
@warpech about the problem that you posted above, I talked about it with @joozek78 and he's aware of that problem and AFA he told me there is nothing he can do about it.

About tests I was thinking that those are flaky tests that are currently failing randomly but I'll check it.

**Comment by PatrykSzwer** [comment]
I fixed the tests. Merging.

**Pull request merged**